### PR TITLE
tests/nested/manual: add test for grades above signed booting with testkeys

### PR DESCRIPTION
**Pull request by anonymouse64**
This test exercises both the fakestore utilities and generated assertions, but
also that we can boot successfully with model assertions other than dangerous as
all our other tests use grade dangerous.

Because this test needs grades above dangerous, we cannot use unasserted snaps,
and need to use the fakestore to generate various snap-declaration and
snap-revision assertions during the course of assembling the image.

Using fakedevicesvc to get a serial assertion is currently disabled, as snapd is
unable to send a request for generating a serial assertion due to:

```
2020-09-03T21:35:05Z ERROR cannot deliver device serial request: invalid assertion: could not validate model assertion (no matching public key "EAD4DbLxK_kn0gzNCXOs3kd6DeMU3f-L6BEsSEuJGBqCORR0gXkdDxMbOm11mRFu" for signature by "developer1")
```

That will be fixed/enabled in a followup PR.

Also note that this PR is not using cloud-init, that will be another PR after this, this PR is purely just about running fakestore uc20 non-dangerous grade model images.

### CODE REVIEW

File: tests/nested/manual/grade-signed-above-testkeys-boot/task.yaml [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+systems: [ubuntu-20.04-64]
+
+environment:
+  # use tpm + secure boot to get full disk encryption, this is explicitly needed
+  # for grade: secured
+  NESTED_ENABLE_TPM: true
+  NESTED_ENABLE_SECURE_BOOT: true
+
+  # kvm doesn't work with tpm + secure boot on gce unfortunately
+  NESTED_ENABLE_KVM: false
```
Comment (mvo5) [inline tests/nested/manual/grade-signed-above-testkeys-boot/task.yaml:12]: Once we landed #9380 this is no longer needed, yes?

File: tests/nested/manual/grade-signed-above-testkeys-boot/task.yaml [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  setup_fake_store "$NESTED_FAKESTORE_BLOB_DIR"
+  teardown_staging_store
+
+  echo Expose the needed assertions through the fakestore
+  cp "$TESTSLIB"/assertions/developer1.account "$NESTED_FAKESTORE_BLOB_DIR/asserts"
+  cp "$TESTSLIB"/assertions/developer1.account-key "$NESTED_FAKESTORE_BLOB_DIR/asserts"
+
+  # modify and repack gadget snap to add a defaults section and use our own
+  # prepare-device hook to use the fakedevicesvc
+  mkdir extra-snaps
```
Comment (mvo5) [inline tests/nested/manual/grade-signed-above-testkeys-boot/task.yaml:70]: This is used automatically by ubuntu-image?

**Review by mvo5 — APPROVED** [review]
This looks fine, thanks for this.

File: tests/nested/manual/grade-signed-above-testkeys-boot/task.yaml (hunk 1 above)
Comment (anonymouse64) [reply tests/nested/manual/grade-signed-above-testkeys-boot/task.yaml:12]: yes

File: tests/nested/manual/grade-signed-above-testkeys-boot/task.yaml (hunk 2 above)
Comment (anonymouse64) [reply tests/nested/manual/grade-signed-above-testkeys-boot/task.yaml:70]: no, it is used by the nested.sh helpers, see 

https://github.com/snapcore/snapd/blob/5a4ecf8f5e0e51957d943c5259e4c369ab718ca0/tests/lib/nested.sh#L467-L474

File: tests/nested/manual/grade-signed-above-testkeys-boot/task.yaml [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    pc-gadget/meta/gadget.yaml
+
+  # TODO: enable this bit when things are ready to use a testkeys signed model
+  # assertion
+  # copy the prepare-device hook to use our fakedevicesvc
+  # mkdir -p pc-gadget/meta/hooks/
+  # cp prepare-device pc-gadget/meta/hooks/
+
+  nested_secboot_sign_gadget pc-gadget "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
+  snap pack pc-gadget/ extra-snaps/
```
Comment (sergiocazzolato) [inline tests/nested/manual/grade-signed-above-testkeys-boot/task.yaml:95]: could you use nested_get_extra_snaps_path function?

File: tests/nested/manual/grade-signed-above-testkeys-boot/task.yaml (hunk 2 above)
Comment (sergiocazzolato) [inline tests/nested/manual/grade-signed-above-testkeys-boot/task.yaml:70]: could you use nested_get_extra_snaps_path function?

File: tests/nested/manual/grade-signed-above-testkeys-boot/task.yaml [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+  # prepare-device hook to use the fakedevicesvc
+  mkdir extra-snaps
+
+  # Get the snakeoil key and cert for signing gadget assets (shim)
+  KEY_NAME=$(nested_get_snakeoil_key)
+  SNAKEOIL_KEY="$PWD/$KEY_NAME.key"
+  SNAKEOIL_CERT="$PWD/$KEY_NAME.pem"
+
+  snap download --basename=pc --channel="20/edge" pc
+  unsquashfs -d pc-gadget pc.snap
```
Comment (sergiocazzolato) [inline tests/nested/manual/grade-signed-above-testkeys-boot/task.yaml:78]: Is it being deleted? I can't see where the dir pc-gadget is deleted

Comment (anonymouse64) [reply tests/nested/manual/grade-signed-above-testkeys-boot/task.yaml:78]: mmm perhaps not, I can delete it too

**Review by sergiocazzolato — APPROVED** [review]
Thanks

**Comment by anonymouse64** [comment]
I ended up fixing a related bug around the repacked-kernel dir persisting after the uc20_build_initramfs_kernel_snap function had returned, because running multiple nested tests on the same system with this failed because the dir already existed in $NESTED_ASSETS_DIR on the second run

**Pull request merged**